### ISSUE TITLE
Angular Theme always sets to "Light (default)" after restart NR

### ISSUE THREAD

**Issue by midnight1979** [node-red/node-red-dashboard#554]
### What are the steps to reproduce?
1. Set "Angular theme everywhere" in "Site" options.
2. Set in "Angular" tab "Dark" in "Light / Dark" option.
3. Do Deploy
4. Restart NR or refresh admin page or logout / login

### What happens?
Theme sets to "Light (default)" again
### What do you expect to happen?
Theme must stay as it was set before restart NR (DARK)
### Please tell us about your environment:

- [x] Node-RED-Dashboard version:2.19.2
- [x] Node-RED version:1.0.3
- [ ] node.js version:
- [ ] npm version:
- [x] Platform/OS:Linux
- [x] Browser:Chromium

**Comment by midnight1979** [comment]
It is still happening even in 2.19.3 release!
1. Logout
2. Login
3. Do some changes in any flow (even just move some node)
4. Do deploy
Theme will change to Light (default).

**Comment by dceejay** [comment]
Did you reset/ clear your local browser cache before loading dashboard after upgrading ?

**Comment by midnight1979** [comment]
Yes i did!

**Comment by midnight1979** [comment]
strange thing - it happening not always...

**Comment by midnight1979** [comment]
look https://youtu.be/nRnpQ4Jm-4M

**Comment by midnight1979** [comment]
i think this issue must be reopened... problem is not solved.

**Comment by dceejay** [comment]
OK - you'll need to give me more info on how to re-create.
I am in angular only - dark mode. I can stop / restart server - close reopen local browser (or just refresh) - and it comes up dark every time so far.

**Comment by midnight1979** [comment]
> OK - you'll need to give me more info on how to re-create.
> I am in angular only - dark mode. I can stop / restart server - close reopen local browser (or just refresh) - and it comes up dark every time so far.

I posted the link to video where i can produce this situation. Have you see it?

**Comment by dceejay** [comment]
Yes. But I don’t have that flow.

**Comment by midnight1979** [comment]
> Yes. But I don’t have that flow.

Ok, i'll create another one, add some nodes and will try to simulate an issue one more time.

**Comment by odenisenko** [comment]
the same here :(

**Comment by lantheman89** [comment]
Same here. Too inconsistent to figure out the exact steps, but for me it happens after several deployments.

**Comment by qwertangel** [comment]
issue is still present in 2.23.3.

**Comment by dceejay** [comment]
"randomly" is the problem... we need to be able to replicate this so we can get a handle on it.
Are you importing flows/nodes ? Re-deploying ? just restarting ? just refreshing the page ?

**Comment by qwertangel** [comment]
in my case, it was always after redeploy (preceded with some changes in dashboard).

**Comment by qwertangel** [comment]
i tried few times and it seems to happen more when I change property of a widget (e.g. change number of days/data points) in chart and redeploy.

**Comment by midnight1979** [comment]
Problem is still presents in the last version, after some changes and deploying flows - theme can become light but in theme properties it stay "dark", now I'm just switch theme to light and then again to dark, deploying changes and it stay dark. But after some changes in flows can become light again...

**Comment by My-Random-Thoughts** [comment]
Just to add that even on 2.29.1 this is still happening.  NR v 1.3.5
Same issue as everyone above.

**Comment by dceejay** [comment]
any repeatable way to demonstrate this ?

**Comment by My-Random-Thoughts** [comment]
Same steps as @midnight1979 in their first post.

- Set to dark
- Make minor change and deploy (even using Modified Nodes option)
- Refresh page (CTRL+F5) - May need to do this a couple of times
- Observe setting has changed back to light..  

With further testing, just refreshing the admin page does not change dark to light, there needs to be a deploy first.

**Comment by dceejay** [comment]
Have pushed a small fix to master - (not on npm) - would be great if you can try it.
(I still can't replicate it here - but did spot some inconsistencies n saving/restoring that value so hopefully this fixes it)

**Comment by My-Random-Thoughts** [comment]
With some limited testing it seems to be OK now.  I can't reproduce the issue anymore.
Thank you.

**Comment by MountainKing91** [comment]
I'm experiencing the same, with node-red 1.3.5 and latest version of dashboard. When I restart node-red the dashboard become "light" even if the theme remains dark. To fix I have to set the them to light, deploy, set it back to dark and then refresh the theme.

**Comment by dceejay** [comment]
yes - the fix is in master branch - not the latest on npm - well it'll be there soon as I release it.

**Comment by My-Random-Thoughts** [comment]
This is still an issue, and I think it might be due to the ui_chart.  I only say that because every other control switched to "light" mode except for the chart whose axis labels are still in dark mode (white text on light grey background)

**Comment by dceejay** [comment]
Really ? you have to refresh the client after deploy to ensure the changes flow through - Is it still dark after a refresh ?

**Comment by My-Random-Thoughts** [comment]
I'll keep testing, but it did do it a couple of times.  There isn't really a consistent way of reproducing the issue, which I know makes it so much harder to debug :)

**Comment by My-Random-Thoughts** [comment]
I can sort of reproduce the issue now using just the buttons on the Layout/Tabs and Links side panel.
Ensure you are set to use the dark theme and open the dashboard page in another tab.

- Click the `+ Tab` button to create one or more new tabs
- From the new tabs click the `+ Group` button a few times (I did 7) to create a bunch of new groups.
- Edit one or more of the groups, change the name or tick the options
- Deploy
- Delete the Tab(s)
- Delete the groups from the `Configuration` / `Unused` tab
- Deploy

It doesn't happen every time and but it's a more reliable method than anything else that I have tried.

**Comment by dceejay** [comment]
hmm - I don't have the knack.. tried several times... still stays dark for me.
